### PR TITLE
LibWeb: Implement <a> element activation behavior for ismap <img>s

### PR DESCRIPTION
**Pull request by shannonbooth**

### CODE REVIEW

File: Userland/Libraries/LibWeb/HTML/HTMLAnchorElement.cpp [hunk 1]
```diff
@@ -61,23 +65,30 @@ void HTMLAnchorElement::activation_behavior(Web::DOM::Event const& event)
...
+
+        // 3. If x is negative, set x to 0.
+        x = max(x, 0);
+
+        // 4. If y is negative, set y to 0.
+        y = max(y, 0);
+
+        // 5. Set hyperlinkSuffix to the concatenation of U+003F (?), the value of x expressed as a base-ten integer using ASCII digits,
+        //    U+002C (,), and the value of y expressed as a base-ten integer using ASCII digits.
+        hyperlink_suffix = MUST(String::formatted("?{},{}", x, y));
```
Comment (trflynn89) [inline Userland/Libraries/LibWeb/HTML/HTMLAnchorElement.cpp:90]: Missing the  "base-ten integer" bit. We do this for the `<input>` element:
```c++
CSSPixels x { mouse_event.offset_x() };
CSSPixels y { mouse_event.offset_y() };

m_selected_coordinate = { x.to_int(), y.to_int() };
```

File: Tests/LibWeb/Text/input/navigation/ismap-img-sends-query-string-from-link.html [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+            println(`Search query from iframe: #${count} = '${new URL(event.data).search}'`);
+            if (count === 1) {
+                internals.click(50, 50);
+            } else if (count === 2) {
+                done();
+            }
+        }
+
+        window.addEventListener("message", handleMessage);
+
```
Comment (trflynn89) [inline Tests/LibWeb/Text/input/navigation/ismap-img-sends-query-string-from-link.html:18]: if you run `prettier` on the .html files, it'll clean up extra whitespace like this

File: Tests/LibWeb/Text/input/navigation/ismap-img-sends-query-string-from-link.html [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    });
+</script>
+<iframe id="testIframe" src=" ../../data/ismap-img-with-link-to-self-sending-url.html" width="400" height="300">
```
Comment (trflynn89) [inline Tests/LibWeb/Text/input/navigation/ismap-img-sends-query-string-from-link.html:21]: works fine probably, but you have an extra space after the opening `src` quote

**Comment by trflynn89** [comment]
I'm concerned the CI failures were legit so I'm rerunning those. That's sort of what I recall seeing with the iframe nav test I tried writing - it caused *other* tests to fail.

**Comment by shannonbooth** [comment]
> I'm concerned the CI failures were legit so I'm rerunning those. That's sort of what I recall seeing with the iframe nav test I tried writing - it caused _other_ tests to fail.

You're right :( It looks real. I wasn't able to see it locally, but maybe I have just been lucky. `input-blur.html` seems to have failed both times on MacOS CI, maybe the OS is somehow important here. Unforuntately I don't have a Mac to test that on though

**Comment by shannonbooth** [comment]
> I'm concerned the CI failures were legit so I'm rerunning those. That's sort of what I recall seeing with the iframe nav test I tried writing - it caused _other_ tests to fail.

Yup, with the test removed the other tests are passing

**Comment by trflynn89** [comment]
> You're right :( It looks real. I wasn't able to see it locally, but maybe I have just been lucky. `input-blur.html` seems to have failed both times on MacOS CI, maybe the OS is somehow important here. Unforuntately I don't have a Mac to test that on though

Might also be timing related with ASAN enabled? Not sure

**Pull request merged**